Modules/@babylonjs/react-native-basekit/scripts/install.js

```diff
@@ -0,0 +1,227 @@
+  // get Babylon React Native package version
+  const packageJsonPath = process.env.npm_package_json;
+  const packageJson = require(packageJsonPath);
+  if (!packageJson.version) {
+    console.err("Babylon React Native version not found");
+    process.exit(1);
+  }
+  // get user project react-native version
+  const projectPackageJsonPath =
+    process.env.npm_config_local_prefix + "/package.json";
+
+  // Read the project's package.json file
+  const projectPackageJson = require(projectPackageJsonPath);
+  // Check if the project's package.json has a 'react-native' dependency
+  if (
+    projectPackageJson.dependencies &&
+    projectPackageJson.dependencies["react-native"]
+  ) {
+    const reactNativeVersion = projectPackageJson.dependencies["react-native"];
+    console.log(`Detected React Native version: ${reactNativeVersion}`);
+
+    const [_, packageMinor] = reactNativeVersion.split('.').map(Number);
+
+    reactNativePostfix = "";
+    if (packageMinor >= "71") {
+      reactNativePostfix = "0.71";
+    } else if (packageMinor >= "70") {
+      reactNativePostfix = "0.70";
+    } else if (packageMinor >= "69") {
+      reactNativePostfix = "0.69";
+    } else {
+      console.error("Unsupported react native version.");
+      process.exit(1);
+    }
+    console.log(
+      `Using Babylon React Native version ${reactNativePostfix} from Package version ${packageJson.version}.`
+    );
+
+    const jsArchive = "react-native.tar.gz";
+    const iosAndroidArchive = `iOSAndroid${reactNativePostfix}.tar.gz`;
+    const windowsArchive = `Windows${reactNativePostfix}.tar.gz`;
+
+    const jsArchiveCachedPath = getCached(packageJson, jsArchive);
+    const iosAndroidArchiveCachedPath = getCached(
+      packageJson,
+      iosAndroidArchive
+    );
+    const windowsArchiveCachedPath = getCached(packageJson, windowsArchive);
+
+    const jsArchiveURL = getBinaryUrl(packageJson, jsArchive);
+    const iOSAndroidArchiveURL = getBinaryUrl(packageJson, iosAndroidArchive);
+    const windowsArchiveURL = getBinaryUrl(packageJson, windowsArchive);
+
+    const reactNativeWindowsVersion =
+      projectPackageJson.dependencies["react-native-windows"];
+    installZip(jsArchiveCachedPath, jsArchiveURL)
+      .then((result) => {
+        installZip(iosAndroidArchiveCachedPath, iOSAndroidArchiveURL)
+          .then((result) => {
+            if (reactNativeWindowsVersion) {
+              installZip(windowsArchiveCachedPath, windowsArchiveURL)
+                .then((result) => {})
+                .catch((err) => {
+                  reject(err);
+                });
+            }
+          })
+          .catch((err) => {
+            reject(err);
+          });
+      })
+      .catch((err) => {
+        console.error("Error install BabylonReactNative archive:", err);
+        process.exit(1);
+      });
+  } else {
+    console.error("No react-native version found for BabylonReactNative.");
+    process.exit(1);
+  }
+}
+
+install();
```

Comment:
Isn't this a fire and forget async call? Will the npm install somehow wait for this operation to complete, or could this still be running after the `npm install` thinks it finishes? I thought I read somewhere it is possible for npm post install scripts to be async...

Reply:
In practice, it looks like npm waits for the install to finish. Can you confirm @RaananW ?